test/testdrive/kafka-time-offset.td

```diff
@@ -304,3 +304,37 @@ cherry    3
 text      mz_offset
 -------------------
 cherry    3
+
+# Make sure that we don't fetch any messages that we don't want to fetch
+
+$ kafka-create-topic topic=t4 partitions=1
+
+$ kafka-ingest format=bytes topic=t4 timestamp=1
+apple
+pie
+
+> CREATE MATERIALIZED SOURCE verify_no_fetch
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-t4-${testdrive.seed}'
+  WITH (kafka_time_offset = -1)
```

Comment:
this test would be slightly more clear for me if we could specify `start_offsets` directly, or maybe with a comment to indicate why that's not possible.

Reply:
The glib answer would be: because I'm testing the `kafka_time_offset` feature. The better answer is that I came to the bug from that angle but will now also add a test that uses `start_offset` in `kafka-avro-sources.td` because you made me realize that I should. :sweat_smile:  That duplicates testing a bit, because the same mechanism is used underneath, but maybe that's okay. What do you think @philip-stoev?

I'll also add a comment here, saying that a time offset of `-1` means we start from the end.

Reply:
The more tests, the better. I personally have nothing against duplication in testing -- no two tests are alike even if one thinks they are, so they will usually hit different issues.